tests/unittests/GraphOptzTest.cpp

```diff
@@ -2210,6 +2210,34 @@ TEST_F(GraphOptz, mergeBANodes) {
   EXPECT_EQ(countNodeKind(F_, Kinded::Kind::BatchedAddNodeKind), 1);
 }
 
+// Check that we are able to replace
+// Add(I, tile(B)) with -> BatchedAdd(I, B).
+TEST_F(GraphOptz, FoldTileAddIntoBatchedAdd) {
+  auto *batch =
+      mod_.createPlaceholder(ElemKind::FloatTy, {2, 3, 3}, "batch", false);
+  auto *added =
+      mod_.createPlaceholder(ElemKind::FloatTy, {1, 3, 3}, "added", false);
+  auto *addedTiled =
+      F_->createTile("addedTiled", added, 2, 0);
+  auto *result =
+      F_ -> createAdd("add", batch, addedTiled);
+  F_ -> createSave("save", result);
+
+  EXPECT_EQ(countNodeKind(F_, Kinded::Kind::TileNodeKind), 1);
+  EXPECT_EQ(countNodeKind(F_, Kinded::Kind::AddNodeKind), 1);
+  EXPECT_EQ(countNodeKind(F_, Kinded::Kind::BatchedAddNodeKind), 0);
+
+  ASSERT_TRUE(F_ -> verify());
+  ::glow::optimize(F_, CompilationMode::Infer);
+  ASSERT_TRUE(F_ -> verify());
+
+  //Check that the Tile node and the Add node is replaced by
+  //a BatchedAdd node.
+  EXPECT_EQ(countNodeKind(F_, Kinded::Kind::TileNodeKind), 0);
+  EXPECT_EQ(countNodeKind(F_, Kinded::Kind::AddNodeKind), 0);
+  EXPECT_EQ(countNodeKind(F_, Kinded::Kind::BatchedAddNodeKind), 1);
+}
```

Comment:
You should verify that the BatchedAdd is correct here, do we usually compile & run the resultant function?

Reply:
I fixed the issue you pointed above. Please take a look.

Reply:
The GraphOptz tests usually just verify that the optimized Graph is the expected one.  I guess the theory is we should have another OperatorTest that verifies that the operators actually do what we expect.

Reply:
@bertmaher I'm fine with that only if we very that the expected one is correct - e.g. has the right inputs and outputs. It's not enough to just replace Tile+Add with BatchedAdd if they wouldn't produced the right output.